src/components/RequestModal/MovieRequestModal.tsx

```diff
@@ -206,60 +199,62 @@ const MovieRequestModal: React.FC<RequestModalProps> = ({
     }
   };
 
-  const isOwner = activeRequest
-    ? activeRequest.requestedBy.id === user?.id
-    : false;
+  if (editRequest) {
+    const isOwner = editRequest.requestedBy.id === user?.id;
+    const showEditButton = hasPermission(
+      [Permission.MANAGE_REQUESTS, Permission.REQUEST_ADVANCED],
+      {
+        type: 'or',
+      }
+    );
 
-  if (activeRequest?.status === MediaRequestStatus.PENDING) {
     return (
       <Modal
         loading={!data && !error}
         backgroundClickable
         onCancel={onCancel}
         title={intl.formatMessage(
           is4k ? messages.pending4krequest : messages.pendingrequest,
-          {
-            title: data?.title,
-          }
+          { title: data?.title }
         )}
-        onOk={() => (isOwner ? cancelRequest() : updateRequest())}
+        onOk={() => (showEditButton ? updateRequest() : cancelRequest())}
         okDisabled={isUpdating}
         okText={
-          isOwner
-            ? isUpdating
-              ? intl.formatMessage(globalMessages.canceling)
-              : intl.formatMessage(messages.cancel)
-            : intl.formatMessage(globalMessages.edit)
+          showEditButton
+            ? intl.formatMessage(messages.edit)
+            : intl.formatMessage(messages.cancel)
+        }
+        okButtonType={showEditButton ? 'primary' : 'danger'}
+        onSecondary={
+          isOwner && showEditButton ? () => cancelRequest() : undefined
+        }
+        secondaryDisabled={isUpdating}
```

Comment:
we should probably make the secondary style here `danger` for when canceling the request is possible?